src/vs/workbench/contrib/testing/browser/explorerProjections/hierarchalByLocation.ts

```diff
@@ -153,9 +153,9 @@ export class HierarchicalByLocationProjection extends Disposable implements ITes
 						break;
 					}
 
-					// parent needs to be re-rendered on an expand update, so that its
+					// parent needs to be re-rendered on an expand update or its children have changed order, so that its
 					// children are rewritten.
-					const needsParentUpdate = existing.test.expand === TestItemExpandState.NotExpandable && patch.expand;
+					const needsParentUpdate = (existing.test.expand === TestItemExpandState.NotExpandable && patch.expand) || patch.item?.sortText;
```

Comment:
I think the fix for error/uri is okay, but changes to the sortText should ideally result in calls to `tree.resort()` rather rerendering the parent.

Reply:
Not sure about the uri change actually, as it's a readonly field. I am working on the resort change- specifically improving the tests to also test the TreeSorter implementation.